### PR TITLE
update butchery to use requirements

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary
```SUMMARY: Features "update butchery to use requirements"```

#### Purpose of change
Moving data from hardcode into json, and also making the messaging clearer when tools are missing.

#### Describe the solution
use a `requirement` for various parts of butchery:
field dress, quarter, skin, dismember: `field_dress`
dissect: `dissect_large`, `dissect_small`
full: `full_butchery_large`, `butchery_small`
quick: `butchery_large`, `butchery_small`

Additionally, uses the same 6 tile range as crafting does.

#### Describe alternatives you've considered
Well we could keep butchery tools hardcoded i guess

#### Testing
spawned in various creates with and without tools.
image

#### Additional context

Inspired by 4c9000b257fccab6ab6b6b93b4767ad0764a5c8c

### CODE REVIEW

File: data/json/requirements/butchery.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+[
+  {
+    "id": "full_butchery_large",
+    "type": "requirement",
+    "//": "omit butcher quality because it is checked separately",
+    "qualities": [
+      { "id": "SURFACE", "level": 3 },
+      { "id": "ROPE", "level": 2 },
+      { "id": "BUTCHERY_RACK", "level": 2 },
```
Comment (Fris0uman) [inline data/json/requirements/butchery.json:9]: Apparently this quality doesn't exist

Comment (KorGgenT) [reply data/json/requirements/butchery.json:9]: lol woops forgot to update it all when i changed the name.

File: data/json/furniture_and_terrain/furniture-tools.json [hunk 2]
```diff
@@ -107,7 +107,7 @@
     "move_cost_mod": -1,
     "coverage": 40,
     "required_str": -1,
-    "crafting_pseudo_item": "rock_forge_pseudo",
```
Comment (I-am-Erk) [inline data/json/furniture_and_terrain/furniture-tools.json:110]: Is this a thing that should be deleted?

**Comment by jbytheway** [comment]
There's a problem with pseudo-items: to the best of my knowledge the player has no way to search for furniture that has a pseudo-item providing a particular quality.  That feels like a necessary step (but maybe it needn't come before this PR is merged).

File: data/json/furniture_and_terrain/furniture-tools.json [hunk 3]
```diff
@@ -107,7 +107,7 @@
...
+    "crafting_pseudo_item": "medium_surface_pseudo",
```
Comment (KorGgenT) [inline data/json/furniture_and_terrain/furniture-tools.json:110]:
```suggestion
    "crafting_pseudo_item": "rock_forge_pseudo",
```

File: data/json/requirements/butchery.json (hunk 1 above)
Comment (KorGgenT) [inline data/json/requirements/butchery.json:9]:
```suggestion
      { "id": "SUSPENDING", "level": 2 },
```

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/butchering-in-0-e/24288/2

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/butchering-in-0-e/24288/4

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/dissecting-in-0-e/24327/7

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/butchering-in-0-e/24288/33

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/butchering-in-0-e/24288/35

**Pull request merged**